code/src/cljs/sixsq/nuvla/ui/notifications/views.cljs

```diff
@@ -0,0 +1,930 @@
+                    :width   2}
+      [:span method-name]]
+     [ui/TableCell {:floated :left
+                    :width   9}
+      [:span [history/link (str "api/" resource-id) resource-id]]]
+     [ui/TableCell {:floated :right
+                    :width   1
+                    :align   :right
+                    :style   {}}
+
+      (when (general-utils/can-edit? notif-subs)
+        [ui/Icon {:name     :cog
+                  :color    :blue
+                  :style    {:cursor :pointer}
+                  :on-click #(dispatch [::events/open-edit-subscription-modal notif-subs false])}])]]))
+
+
+(defn manage-subscriptions-modal
+  []
+  (let [tr (subscribe [::i18n-subs/tr])
+        notif-methods (subscribe [::subs/notification-methods])
+        subscriptions (subscribe [::subs/notification-subscriptions])
+        visible? (subscribe [::subs/notification-subscriptions-modal-visible?])]
+    (dispatch [::events/get-notification-methods])
+    (dispatch [::events/get-notification-subscriptions])
+    (fn []
+      (let [header (@tr [:subscriptions-manage])]
+        [:div]
+        [ui/Modal {:open       @visible?
+                   :close-icon true
+                   :on-close   #(dispatch [::events/close-notification-subscription-modal])}
+
+         [ui/ModalHeader header]
+
+         [ui/ModalContent {:scrolling false}
+          [ui/Table {:style {:margin-top 10}}
+           [ui/TableHeader
+            [ui/TableRow
+             [ui/TableHeaderCell {:content (str/capitalize (@tr [:status]))}]
+             [ui/TableHeaderCell {:content (str/capitalize (@tr [:method]))}]
+             [ui/TableHeaderCell {:content (str/capitalize (@tr [:resource]))}]
+             [ui/TableHeaderCell {:content (str/capitalize (@tr [:action]))}]]]
+           [ui/TableBody
+            (for [sub @subscriptions]
+              ^{:key (:id sub)}
+              [single-notification-subscription sub notif-methods])]]
+          [utils-validation/validation-error-message ::subs/form-valid?]]]))))
+
+
+(defn subs-method-dropdown
+  [current-value notif-methods disabled?]
+  ^{:key current-value}
+  [ui/Dropdown
+   {:selection     true
+    :fluid         false
+    :disabled      disabled?
+    :default-value (if (not (nil? current-value))
+                     current-value
+                     (-> @notif-methods first :id))
+    :on-change     (ui-callback/value
+                     #(do (dispatch-sync [::events/update-subscription :method %])))
+    :options       (map (fn [{id :id, method-name :name}]
+                          {:key id, :value id, :text method-name})
+                        @notif-methods)}])
+
+
+(defn subs-status-dropdown
+  [current-value]
+  ^{:key current-value}
+  [ui/Dropdown
+   {:selection     true
+    :fluid         false
+    :default-value current-value
+    :on-change     (ui-callback/value
+                     #(do (dispatch-sync [::events/update-subscription :status %])))
```

Comment:
try a maximum to not use dispatch-sync. do is useless

Reply:
done